src/main/java/org/mskcc/cbio/portal/scripts/GeneticAlterationIncrementalImporter.java

```diff
@@ -0,0 +1,105 @@
+package org.mskcc.cbio.portal.scripts;
+
+import org.mskcc.cbio.portal.dao.DaoException;
+import org.mskcc.cbio.portal.dao.DaoGeneticAlteration;
+import org.mskcc.cbio.portal.dao.DaoGeneticProfileSamples;
+import org.mskcc.cbio.portal.model.CanonicalGene;
+import org.mskcc.cbio.portal.util.ArrayUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+public class GeneticAlterationIncrementalImporter extends GeneticAlterationImporter {
+
+    private final List<Integer> fileOrderedSampleList;
+    private final DaoGeneticAlteration daoGeneticAlteration = DaoGeneticAlteration.getInstance();
+    private final HashMap<Integer, HashMap<Integer, String>> geneticAlterationMap;
+
+    public GeneticAlterationIncrementalImporter(
+            int geneticProfileId,
+            List<Integer> fileOrderedSampleList
+    ) throws DaoException {
+
+        this.geneticProfileId = geneticProfileId;
+        this.geneticAlterationMap = daoGeneticAlteration.getGeneticAlterationMapForEntityIds(geneticProfileId, null);
+        this.fileOrderedSampleList = fileOrderedSampleList;
+
+        ArrayList <Integer> savedOrderedSampleList = DaoGeneticProfileSamples.getOrderedSampleList(this.geneticProfileId);
+        int initialOrderSampleListSize = savedOrderedSampleList.size();
+        geneticAlterationMap.forEach((geneticEntityId, sampleToValue) -> {
+            if (sampleToValue.size() != initialOrderSampleListSize) {
+                throw new IllegalStateException("Number of samples ("
+                        + sampleToValue.size() + ") for genetic entity with id "
+                        + geneticEntityId + " does not match with the number in the inital sample list ("
+                        + initialOrderSampleListSize + ").");
+            }
+        });
+        // add all new sample ids at the end
+        this.orderedSampleList = new ArrayList<>(savedOrderedSampleList);
+        List<Integer> newSampleIds = this.fileOrderedSampleList.stream().filter(sampleId -> !savedOrderedSampleList.contains(sampleId)).toList();
+        this.orderedSampleList.addAll(newSampleIds);
+        DaoGeneticProfileSamples.deleteAllSamplesInGeneticProfile(this.geneticProfileId);
+        this.storeOrderedSampleList();
```

Comment:
same comment as above (database operations in constructor)

Reply:
addressed in the single commit with the above one https://github.com/cBioPortal/cbioportal-core/pull/48/commits/0edaec3fc67d580dd7ebd30913c50f8ea989de0f

Reply:
👍